docs/LayoutSpecs.json

```diff
@@ -808,7 +808,7 @@
             {
               "text": "Color",
               "iconUrl": "",
-              "elementType": "group",
+              "elementType": "class",
```

Comment:
so when does class get used? When should it be used opposed to group? Can you add these types to the documentation files? / wiki?

Do we ever generate `classes` or `coreGroups` when importing some other library which is not defined in the layout spec?

Reply:
@ramramps also, so you will need to update the layoutspec in revit and dynamo as a consequence of this PR as well to get the same behavior right?

Reply:
First, I understand this work is being done to accomplish the particular design requirements.
I think it would be useful for someone to enumerate the rules that are now encoded in this repo.
I could be wrong, but they seem pretty complicated to me.

I will take a shot:
* Top Level categories display Icons
* groups or classes do not display icons
* NonNested categories do not expand
* Top Level categories expand
* other stuff?

I do not think we will be able to maintain this matrix of interactions without properly documenting it and the new classes/ groups, etc...

This is not a criticism of the implementation here. I am fine with this approach, but would be happier if we did not need to introduce new types to accomplish this, just apply the minimal logic required to existing types throughout the library with no special logic for items in the layout spec - remember we'll need another giant layout spec for TS.

at a minimum let us please document these rules throughly.

Reply:
@mjkkirschner Thanks for the comments. 
1. Dynamo and Revit layoutspec needs a change. 
2. This is definitely to get that particular design. This was easy in WPF. But in librarie.js, everything is dependent on `elementType`.  I don't have a choice other than adding a new one. 
3. Auto expand - as per the task, it is only for top level. Per my understanding from layoutspec, this is the order after the change`section-category-coregroup-group-none`. All the nested category gets `group` by default. 
4.Before the change, the order is `section-category-group-none`. So, there is no clear distinction between the top level and nested categories. Well, we can leave it this way too.
5. I am afraid that if layoutspec is auto generated, then achieving this design will get complicated and requires careful look of how it is made today.
6.I will add a wiki page with rules per my understanding. 

Also, I will rename `coregroup to groups`.  So, the top level has groups of group.

Reply:
@ramramps thanks for the detailed response - have you added the wiki page yet (6) on your list?

Reply:
not yet. the rules are simplified now. I will start wiki , and let @Racel add the rules. She knows better than me.